Cabal/Distribution/Package.hs

```diff
@@ -395,12 +396,20 @@ mkLegacyUnitId = newSimpleUnitId . mkComponentId . display
 data Dependency = Dependency PackageName VersionRange
                   deriving (Generic, Read, Show, Eq, Typeable, Data)
 
+-- | Describes a dependency on an executable from a package
+--
+data ExeDependency = ExeDependency
+                     PackageName
+                     UnqualComponentName -- name of executable component of package
+                     VersionRange
+                     deriving (Generic, Read, Show, Eq, Typeable, Data)
```

Comment:
Sure, doesn't hurt to add it now. Good hint on unqual vs qual for anyone working with the code base.